server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -679,7 +680,7 @@ private static void addGeneratedToVerify(Map<ColumnIdent, ColumnConstraint> colu
      * Looks for new columns in the values of the given IndexItem and returns them.
      */
     public List<Reference> collectSchemaUpdates(IndexItem item) throws IOException {
-        ArrayList<Reference> newColumns = new ArrayList<>();
+        HashSet<Reference> newColumns = new HashSet<>();
```

Comment:
I think we have some tests that assert certain position/ordinals - if we use a set here that may become undeterministic and flaky. 

Probably worth giving tests a few iterations to see if all is good - and if not, use a LinkedHashSet?

Reply:
Good point, thx. Will check.